error.go

```diff
@@ -0,0 +1,41 @@
+package uipath
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+// RequestError defines how the error looks like from the response
+type RequestError struct {
+	Message          string `json:"message"`
+	ErrorName        string `json:"error"`
+	ErrorCode        int    `json:"errorCode"`
+	ErrorDescription string `json:"error_description"`
+	TraceIDs         string `json:"traceId"`
+	ResourceIds      []uint `json:"resourceIds"`
+}
+
+func (r *RequestError) Error() string {
+	if r.ErrorName != "" {
+		return fmt.Sprintf("Request Failed: Error Code(%s) %s", r.ErrorName, r.ErrorDescription)
+	}
+
+	return fmt.Sprintf("Request Failed: Error Code(%d) %s", r.ErrorCode, r.Message)
+}
+
+//ErrorResponseHandler handles the errors from the uipath response
+func ErrorResponseHandler(statusCode int, errResp []byte) error {
+	var requestError RequestError
+
+	if len(errResp) < 1 {
```

Comment:
How exactly this works? It is kind of not clear to me 🤔  Do you mind adding a comment?

Reply:
sure! basically it just checks the response body if it's empty. If it is, we assume that it's an HTTP error from the server